src/Digdir.Domain.Dialogporten.WebApi/Common/Authentication/TokenIssuerCache.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Protocols;
+using System.Collections.ObjectModel;
+
+namespace Digdir.Domain.Dialogporten.WebApi.Common.Authentication;
+
+public interface ITokenIssuerCache
+{
+    public Task<string?> GetIssuerForScheme(string schemeName);
+}
+
+public class TokenIssuerCache : ITokenIssuerCache, IDisposable
+{
+    private readonly Dictionary<string, string> _issuerMappings = new();
+    private readonly SemaphoreSlim _initializationSemaphore = new(1, 1);
+    private bool _initialized;
+    private readonly IEnumerable<JwtBearerTokenSchemasOptions>? _jwtTokenSchemas;
+
+    public TokenIssuerCache(IConfiguration configuration)
+    {
+        _jwtTokenSchemas = configuration
+            .GetSection(WebApiSettings.SectionName)
+            .Get<WebApiSettings>()
+            ?.Authentication
+            ?.JwtBearerTokenSchemas;
+    }
```

Comment:
```suggestion
    public TokenIssuerCache(IOptions<WebApiSettings> apiSettings)
    {
        _jwtTokenSchemas = apiSettings
            .Value
            .Authentication
            .JwtBearerTokenSchemas
            ?? throw new ArgumentException("JwtBearerTokenSchemas is required.");
    }
```

 Options blir validert.